packages/zoe/src/zoeService/instanceAdminStorage.js

```diff
@@ -1,49 +1,284 @@
 import { E } from '@endo/eventual-send';
-import { makeWeakStore } from '@agoric/store';
+import {
+  canBeDurable,
+  makeScalarBigSetStore,
+  provideDurableWeakMapStore,
+  vivifyKindMulti,
+  vivifyFarClassKit,
+  M,
+  provide,
+} from '@agoric/vat-data';
+import { makeZoeSeatAdminFactory } from './zoeSeat.js';
+import { defineDurableHandle } from '../makeHandle.js';
+import { InstanceAdminI, InstanceHandleShape } from '../typeGuards.js';
 
-/**
- *
- */
-export const makeInstanceAdminStorage = () => {
-  /** @type {WeakStore<Instance,InstanceAdmin>} */
-  const instanceToInstanceAdmin = makeWeakStore('instance');
+const { quote: q, details: X } = assert;
 
-  /** @type {import('./utils.js').GetPublicFacet} */
-  const getPublicFacet = async instanceP =>
-    E.when(instanceP, instance =>
-      instanceToInstanceAdmin.get(instance).getPublicFacet(),
-    );
+const InstanceAdminStorageShape = M.interface('InstanceAdminStorage', {
+  getPublicFacet: M.call(M.eref(InstanceHandleShape)).returns(M.promise()),
+  getBrands: M.call(InstanceHandleShape).returns(M.promise()),
+  getIssuers: M.call(InstanceHandleShape).returns(M.promise()),
+  getTerms: M.call(InstanceHandleShape).returns(M.promise()),
+  getOfferFilter: M.call(M.eref(InstanceHandleShape)).returns(M.promise()),
+  setOfferFilter: M.call(M.eref(InstanceHandleShape)).returns(),
+  getInstallationForInstance: M.call(InstanceHandleShape).returns(M.promise()),
+  getInstanceAdmin: M.call(InstanceHandleShape).returns(InstanceAdminI),
+  initInstanceAdmin: M.call(InstanceHandleShape).returns(M.promise()),
+  deleteInstanceAdmin: M.call(InstanceHandleShape).returns(M.promise()),
+});
 
-  /** @type {GetBrands} */
-  const getBrands = async instance =>
-    instanceToInstanceAdmin.get(instance).getBrands();
+/** @param {import('@agoric/vat-data').Baggage} baggage */
+export const makeInstanceAdminStorage = baggage => {
+  const makeIAS = vivifyFarClassKit(
+    baggage,
+    'InstanceAdmin',
+    InstanceAdminStorageShape,
+    () => ({
+      instanceToInstanceAdmin: provideDurableWeakMapStore(
+        baggage,
+        'instanceToInstanceAdmin',
+      ),
+    }),
+    {
+      // ZoeStorageManager uses the accessor facet to get info about instances
+      accessor: {
+        async getPublicFacet(instanceP) {
+          const { state } = this;
+          return E.when(instanceP, async instance => {
+            const ia = state.instanceToInstanceAdmin.get(instance);
+            return ia.getPublicFacet();
+          });
+        },
```

Comment:
Pleasingly short continuation of the example above.

```suggestion
        async getPublicFacet(instance) {
          const { state } = this;
          const ia = state.instanceToInstanceAdmin.get(instance);
          return ia.getPublicFacet();
        },
```

Reply:
done